README.md

```diff
@@ -1,10 +1,102 @@
 # kv-go
-Distributed in-memory (at least for now) KV storage written in Go
 
-#Motivation
-Just For Fun
-This is an educational project about how to develop scalable distributed systems.
-
-#Disclaimer
+# Disclaimer
 Use this project at your own risk.
 PRs are welcome!
+
+## Distributed in-memory (at least for now) KV storage written in Go
+
+## Project Overview
+
+### Objective
+
+To design and implement a distributed in-memory key-value storage system using the Go programming language. The system should be capable of handling high throughput, be fault-tolerant, and easily scalable. This is an educational project about how to develop scalable distributed systems. Just for fun.
+
+
+### Scope
+
+1. In-Memory Key-Value Storage
+2. Data Replication
+3. Data Sharding
+4. Fault Tolerance
+5. Consistency Guarantees
+6. API for CRUD operations
+7. Monitoring and Logging
+
+## System Architecture
+
+### Components
+
+1. **Node**: Basic unit of the storage system, responsible for storing in-memory data.
+2. **Cluster Manager**: Manages the nodes in the system.
+3. **Client Library**: Used for interacting with the storage system.
+4. **Monitoring Service**: Monitors system health and performance.
+
+### Data Flow
+
+1. The client sends a request via the client library.
+2. The client library communicates with the Cluster Manager to find the appropriate node.
+3. The node performs the operation and returns the result to the client library.
+4. The client library returns the result to the client.
```

Comment:
Highly encourage you to add cli app to the list of goals for the whole project. It will help in understanding how SDK should look like, and how cli app should be built.
And yes, cli is just easier for interacting with db.